ui/plugin-system/src/model/variables.ts

```diff
@@ -18,29 +18,30 @@ import { usePlugin } from '../components/PluginLoadingBoundary';
  * Plugin for handling custom VariableDefinitions.
  */
 export interface VariablePlugin<Options extends JsonObject = JsonObject> {
-  useVariableOptions: UseVariableOptionsHook<Options>;
+  getVariableOptions: GetVariableOptions<Options>;
```

Comment:
nit, maybe a comment here could be useful on why this is promise-based instead of a hook or it can wait as long as we document that decision later when we add actual plugin docs.